File: backend/models/challenge.ts
import { Schema, model } from "mongoose";
import { ObjectId } from "mongodb";

export const challengeSchema = new Schema({
  title: { type: String, required: true },
  status: { type: String, default: "open" },
  posts: { type: [ObjectId], ref: "post", default: [] },
});

const challengeModel = model("challenge", challengeSchema);
export default challengeModel;

File: flutter/lib/models/Challenge.dart
import 'package:cooking/models/Post.dart';

class Challenge {
  final String title;
  final String status;
  final List<dynamic> posts;

  Challenge({required this.title, required this.status, required this.posts});

  Challenge.fromJson(json, {bool isPopulated = false})
      : status = json["status"],
        title = json["title"],
        posts = isPopulated
            ? json["posts"]
                .map((postJson) => Post.fromJson(postJson, isPopulated: true))
                .toList()
            : [];
}
